src/internal/pjsdb/sanitize.go

```diff
@@ -0,0 +1,64 @@
+package pjsdb
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/pachsql"
+	"github.com/pachyderm/pachyderm/v2/src/internal/pctx"
+	"github.com/pachyderm/pachyderm/v2/src/internal/storage/fileset"
+)
+
+// sanitize.go contains a sanitization pattern to decouple the CRUD API from *fileset.Storage and validate
+// CreateJob() parameters.
+
+// CreateJobRequest is a bundle of related fields required for a CreateJob() invocation.
+// In pfsdb, we used a pattern of forcing the caller to convert their resources to a _Info struct, but its problematic since
+// each database function only really needs a subset of the fields and it is not clear to the caller which fields are required.
+// CreateJobRequest cannot be used directly, it has to be first sanitized via Sanitize()
+type CreateJobRequest struct {
+	Parent      JobID
+	Input, Spec string
+}
+
+type createJobRequest struct {
+	parent      sql.NullInt64
+	input, spec string
+	specHash    []byte
+}
+
+// Sanitize should be invoked prior to calling CreateJob() to ensure that a job can be properly created.
+// this functionally is separate from the CreateJob() so that CreateJob() doesn't require access to a *fileset.Storage
```

Comment:
Why _not_ pass a `*fileset.Storage` to `CreateJob`?

Reply:
My strongest argument is that the storage layer is only needed for validation for this specific function. We just want to check if the fileset points to something that actually exists. The core functionality doesn't dependent on it at all. My other argument is that it _feels_ like the two APIs (this and the storage API) should be logically separate and the caller should be making them interoperate. Lastly, we're planning to eventually sunset the current storage.filesets table and remodel that so that fileset IDs are hashes of their root (and the tree structures would be convergent to ensure that two filesets with the same contents have the same hash). I figured that structuring this way would make it easier to update the validation code later.

Reply:
> the storage layer is only needed for validation for this specific function

But this specific function is required to be called prior to calling `CreateJob`, right? So the storage layer _is_ required by `CreateJob`.

When things change later the refactor should be mechanical, right?

Reply:
If you didn't have validation it wouldn't be required. You could force the caller to validate the parameters before passing down to the create JobAPI. That's what this does.